packages/lodestar/test/unit/executionEngine/httpRetry.test.ts

```diff
@@ -0,0 +1,282 @@
+import chai, {expect} from "chai";
+import chaiAsPromised from "chai-as-promised";
+import {fastify} from "fastify";
+import {AbortController} from "@chainsafe/abort-controller";
+import {fromHexString} from "@chainsafe/ssz";
+import {ExecutionEngineHttp, parseExecutionPayload, serializeExecutionPayload} from "../../../src/executionEngine/http";
+import {defaultExecutionEngineHttpOpts} from "../../../lib/executionEngine/http";
+import {bytesToData, numToQuantity} from "../../../src/eth1/provider/utils";
+
+chai.use(chaiAsPromised);
+
+describe("ExecutionEngine / http ", () => {
+  const afterCallbacks: (() => Promise<void> | void)[] = [];
+  after(async () => {
+    while (afterCallbacks.length > 0) {
+      const callback = afterCallbacks.pop();
+      if (callback) await callback();
+    }
+  });
+
+  let executionEngine: ExecutionEngineHttp;
+  let returnValue: unknown = {};
+  let reqJsonRpcPayload: unknown = {};
+  let baseUrl: string;
+  let errorResponsesBeforeSuccess = 0;
+  let controller: AbortController;
+
+  before("Prepare server", async () => {
+    controller = new AbortController();
+    const server = fastify({logger: false});
+
+    server.post("/", async (req) => {
+      if (errorResponsesBeforeSuccess === 0) {
+        reqJsonRpcPayload = req.body;
+        delete (reqJsonRpcPayload as {id?: number}).id;
+        return returnValue;
+      } else {
+        --errorResponsesBeforeSuccess;
+        throw Error(`Will succeed after ${errorResponsesBeforeSuccess} more attempts`);
+      }
+    });
+
+    afterCallbacks.push(async () => {
+      controller.abort();
+      await server.close();
+    });
+
+    baseUrl = await server.listen(0);
+
+    executionEngine = new ExecutionEngineHttp(
+      {
+        urls: [baseUrl],
+        retryAttempts: defaultExecutionEngineHttpOpts.retryAttempts,
+        retryDelay: defaultExecutionEngineHttpOpts.retryDelay,
+      },
+      controller.signal
+    );
+  });
+
+  describe("getPayload", async () => {
+    it("getPayload with retry", async function () {
+      this.timeout("10 min");
+      /**
+       * curl -X POST -H "Content-Type: application/json" --data '{"jsonrpc":"2.0","method":"engine_getPayload","params":["0x0"],"id":67}' http://localhost:8545
+       */
```

Comment:
you don't need to rewrite all the execution api cases, as it will be difficult to maintain/update any changes in two places, rather just test out `fetchWithRetries` functionality exhaustively against any hypothetical request response (in the same way its happening here) basically write its test cases independentaly of the execution api logic

Reply:
What do you say to removing the tests as you suggested but leaving just:
 
- `notifyForkchoiceUpdate no retry when no pay load attributes` and 
- `notifyForkchoiceUpdate with retry when pay load attributes` tests? 

as this are specific to how the node calls the EL, and when those calls should be retried or not.

Reply:
ok, lets leave it at that, no need to remove :+1: